components/contract-components/contract-release-form/index.tsx

```diff
@@ -184,561 +146,197 @@ export const ContractReleaseForm: React.FC<ContractReleaseFormProps> = ({
       ?.implementationInitializerFunction || "initialize",
   );
 
-  const deployParams = watch("isDeployableViaProxy")
-    ? initializerParams
-    : constructorParams;
+  const deployParams =
+    contractSelection === "proxy" ? initializerParams : constructorParams;
 
   // during loading and after success we should stay in loading state
   const isLoading = publishMutation.isLoading || publishMutation.isSuccess;
+
+  useEffect(() => {
+    window?.scrollTo({
+      top: 0,
+      behavior: "smooth",
+    });
+  }, [inputGroupToShow]);
+
   return (
-    <Card w="100%" p={{ base: 6, md: 10 }}>
-      <Flex
-        as="form"
-        id="contract-release-form"
-        onSubmit={handleSubmit((data) => {
-          // the drawer has another form inside it which triggers this one on submit
-          // hacky solution to avoid double submission
-          if (isDrawerOpen) {
-            return;
-          }
-          trackEvent({
-            category: "publish",
-            action: "click",
-            label: "attempt",
-            uris: contractId,
-            release_id: `${ensNameOrAddress}/${publishMetadata.data?.name}`,
-          });
-          publishMutation.mutate(
-            {
-              predeployUri: contractId,
-              extraMetadata: data,
-              contractName: publishMetadata.data?.name,
-            },
-            {
-              onSuccess: () => {
-                onSuccess();
-                trackEvent({
-                  category: "publish",
-                  action: "click",
-                  label: "success",
-                  uris: contractId,
-                  release_id: `${ensNameOrAddress}/${publishMetadata.data?.name}`,
-                  version: data.version,
-                  is_proxy: data.isDeployableViaProxy,
-                  is_factory: data.isDeployableViaFactory,
-                });
-                if (successRedirectUrl) {
-                  router.push(
-                    successRedirectUrl,
-                    undefined,
-                    // reset scroll after redirect
-                    // shallow render (aka do not wait for SSR)
-                    { scroll: true, shallow: true },
-                  );
-                }
+    <FormProvider {...form}>
+      <Box w="100%">
+        <Flex
+          as="form"
+          id="contract-release-form"
+          onSubmit={form.handleSubmit((data) => {
```

Comment:
JSX would be more legible if we pull it out to a function above instead of inlining it in JSX

Reply:
we don't do that anywhere because that way we lose the inferred typing

Reply:
We are not leveraging the inferred typing here